backend/compact-connect/lambdas/cloudfront-csp/index.js

```diff
@@ -1,5 +1,337 @@
+
+// ============================================================================
+//                                   HELPERS                                  =
+// ============================================================================
+/**
+ * Get the request domain from the lambda event record.
+ * @param  {object} eventRecord The cloudfront record from the lambda event.
+ * @return {string}             The bare domain of the request domain.
+ */
+const getRequestDomain = (eventRecord) => {
+    let requestDomain = environments.csg.prod.webFrontend;
+
+    if (eventRecord) {
+        const requestHostHeaderValue = eventRecord.request?.headers?.host?.[0]?.value;
+
+        if (requestHostHeaderValue) {
+            requestDomain = requestHostHeaderValue;
+        }
+    }
+
+    return requestDomain;
+};
+
+/**
+ * Get a fully qualified domain URI with the protocol scheme.
+ * @param  {string} domain The bare domain string.
+ * @return {string}        The fully-qualified domain string.
+ */
+const getFullyQualified = (domain) => {
+    const protocol = 'https://';
+    let fullyQualified = '';
+
+    if (domain && typeof domain === 'string' && !domain.startsWith(protocol)) {
+        fullyQualified = `${protocol}${domain}`;
+    }
+
+    return fullyQualified;
+};
+
+/**
+ * Helper to get the fully-qualified domains for connected services based on the request domain.
+ * @param  {string} requestDomain The bare domain of the request.
+ * @return {object}               A map of fully-qualified domains for the request environment.
+ *   @return {string} dataApi       The data API fully-qualified domain.
+ *   @return {string} s3Upload      The S3 fully-qualified domain for uploading state files.
+ */
+const getEnvironmentUrls = (requestDomain) => {
+    const environmentUrls = {};
+    let environment;
+
+    switch (requestDomain) {
+    case environments.csg.prod.webFrontend:
+        environment = environments.csg.prod;
+        break;
+    case environments.csg.test.webFrontend:
+        environment = environments.csg.test;
+        break;
+    case environments.ia.prod.webFrontend:
+        environment = environments.ia.prod;
+        break;
+    case environments.ia.test.webFrontend:
+        environment = environments.ia.test;
+        break;
+    case environments.ia.justin.webFrontend:
+        environment = environments.ia.justin;
+        break;
+    default:
+        environment = environments.csg.prod;
+        break;
+    }
+
+    environmentUrls.dataApi = getFullyQualified(environment.dataApi);
+    environmentUrls.s3Upload = getFullyQualified(environment.s3Upload);
+
+    return environmentUrls;
+};
+
+/**
+ * Helper to escape CSP src keywords.
+ * @param  {string} keyword The standard keyword value.
+ * @return {string}         The escaped keyword value.
+ */
+const srcKeywordEscape = (keyword) => {
+    let escaped = '';
+
+    if (keyword && typeof keyword === 'string') {
+        escaped = `'${keyword}'`.toLowerCase();
```

Comment:
how is lowercasing it "escaping" it?

Reply:
The quotes around `${keyword}` are doing the escaping. The lowercasing is just keeping things uniform.